src/main/java/store/model/service/PurchaseProcessing.java

```diff
@@ -0,0 +1,143 @@
+package store.model.service;
+
+import camp.nextstep.edu.missionutils.DateTimes;
+import java.time.chrono.ChronoLocalDate;
+import store.model.domain.Cart;
+import store.model.domain.Product;
+import store.model.domain.Promotion;
+import store.model.domain.PromotionList;
+import store.model.domain.Receipt;
+import store.model.domain.Stock;
+import store.view.InputCallback;
+
+public class PurchaseProcessing {
+    private final PromotionList promotionList;
+    private final InputCallback inputCallback;
+    private final Receipt receipt;
+    private final Stock stock;
+    private final Cart cart;
+
+    public PurchaseProcessing(PromotionList promotionList, InputCallback inputCallback, Receipt receipt, Stock stock,
+                              Cart cart) {
+        this.promotionList = promotionList;
+        this.inputCallback = inputCallback;
+        this.receipt = receipt;
+        this.stock = stock;
+        this.cart = cart;
+    }
+
+    public void applyCart() {
+        for (Product product : cart.getProducts()) {
+            if (isApplicableTo(product.getName())) {
+                apply(product);
+            } else {
+                receipt.addProduct(product);
+                stock.purchase(product.getName(), product.getQuantity());
+            }
+        }
+    }
+
+    private void apply(Product product) {
+        String productName = product.getName();
+        int cartProductQuantity = product.getQuantity();
+        int stockPromotionProductQuantity = stock.findPromotionProductByName(productName).getQuantity();
+        Promotion promotion = promotionList.findPromotionByName(
+                stock.findPromotionProductByName(productName).getPromotionName());
+        int promotionUnit = promotion.getPromotionUnit(); // 프로모션 단위 (2+1 상품 -> 3, 1+1 상품 -> 2)
+        String promotionName = promotion.getPromotionName();
+        int remaining = cartProductQuantity % promotionUnit; // 나머지 (구매량을 프로모션 단위로 나눈 나머지)
+        int cleanCartProductQuantity = cartProductQuantity + promotionUnit - remaining; // 구매량 + 나머지
+        int purchasePromotionBundle = 0; // 프로모션 단위로 구매한 묶음
+        int priceOfProduct = product.getPrice();
+
+        while (cleanCartProductQuantity > 0 && stockPromotionProductQuantity > 0) {
+            cleanCartProductQuantity -= promotionUnit;
+            stockPromotionProductQuantity -= promotionUnit;
+            purchasePromotionBundle++;
+        }
+
+        if (stockPromotionProductQuantity > 0 && remaining > 0) { // 상품 추가 구매 제안 상황
+            boolean progress = inputCallback.requestPromotionAddProduct(productName, promotionUnit - remaining);
+            addPromotionProductApply(progress, purchasePromotionBundle, promotionUnit, productName, priceOfProduct,
+                    promotionName,
+                    cartProductQuantity);
+        } else if (stockPromotionProductQuantity > 0 && remaining == 0) { // 프로모션 즉시 적용 상황
```

Comment:
공통 피드백 내용에 따라 else if 대신에 if에 return을 사용해서 처리하는 게 좋았을거 같습니다!
```suggestion
                    promotionName,
                    cartProductQuantity);
                    return;
        }  if (stockPromotionProductQuantity > 0 && remaining == 0) { // 프로모션 즉시 적용 상황
```

Reply:
`void`로 선언한 함수에 `return`을 저런식으로 쓰면 되겠네요! 하나 배우고 갑니다.
저도 else 를 쓰고싶지 않았지만 `void`로 선언한 함수에서는 `return`을 못쓴다고 생각했어서 이렇게 해버렸네요...